app/src/main/java/com/mk/tasky/authentication/domain/usecase/ValidEmailUseCase.kt

```diff
@@ -0,0 +1,9 @@
+package com.mk.tasky.authentication.domain.usecase
+
+import android.util.Patterns
+
+class ValidEmailUseCase {
+    operator fun invoke(email: String): Boolean {
+        return Patterns.EMAIL_ADDRESS.matcher(email).matches()
+    }
+}
```

Comment:
Some feedback for this use case class:
1. I'd prefer choosing verbs as name for your use cases, so that it's something you do. For example `ValidateEmailUseCase`. Otherwise, it might raise the expectation that this use case is only for valid emails.
2. `Patterns` is a class from the Android SDK (you can see that at the import statement). Ideally, you should avoid using Android dependencies in domain (also things like context should be avoided) because then you can't write local unit tests for your use cases (if they contain Android specific dependencies, they must be instrumented tests and run on an emulator which is much slower).  Being able to easily write unit tests is in the end one major reason why we use use cases :) 
You can solve this by adding an abstraction for the pattern matching. Something like this.
In domain:
```
interface EmailMatcher {
    fun matches(email: String): Boolean
}
```

Then in data layer:
```
class EmailMatcherImpl: EmailMatcher {
    override fun matches(email: String): Boolean {
        return Patterns.EMAIL_ADDRESS.matcher(email).matches()
    }
}
```

Then, your use case can take an `EmailMatcher` instance in the constructor and use that. Or you directly use the abstraction in your ViewModels to avoid the use case boilerplate.

Reply:
Awesome. Sounds good!